environment/src/main/java/com/sequenceiq/environment/environment/service/network/NetworkMetadataValidationService.java

```diff
@@ -54,13 +54,18 @@ public NetworkMetadataValidationService(CloudNetworkService cloudNetworkService,
     }
 
     private Map<String, CloudSubnet> removePrivateSubnets(Map<String, CloudSubnet> endpointGatewaySubnetMetas) {
-        LOGGER.debug("Removing any private subnets from the provided endpoint gateway list because they won't be used.");
-        if (endpointGatewaySubnetMetas == null || endpointGatewaySubnetMetas.isEmpty()) {
-            return Map.of();
+        if (entitlementService.endpointGatewayValidationEnabled(ThreadBasedUserCrnProvider.getAccountId())) {
```

Comment:
Does it mean that every customer needs to have this entitlement as default?
As the original behaviour was the validation of the subnets.

Reply:
The plan is that no customers will need this. By default we will not do the validation. However, @kmanamcheri suggested that instead of simply deleting the validation logic, we put it behind an entitlement, so that if it turns out it's better to have it than to not we can re-enable it quickly be enabling the entitlement. That's just a failsafe, and I don't anticipate we'll ever have to enable this entitlement.

I'm working on a separate effort to add new metric information around the endpoint gateway logic, so that we can see how customers are really using it and use that to see if we should bring back the validation logic in a different form in a later release. But that's not part of this, because I'm trying to get the initial version finished for Exxon.

Reply:
Re: The original behavior, this logic is only called during initial environment creation. If any existing environments are using this feature, then they'll be past this part of the code already and won't revisit it. If a customer creates a new environment after this change, then we will not do the validation as part of that environment's creation.

Reply:
Yeah but the validation is now disabled by default and the customer needs to ask for an entitlement to have the validation before env creation.
I would prefer to have the original behaviour which was the validation by default and have an entitlement for disabling it. 
@doktoric @lacikaaa @keyki @akanto What do you think?

Reply:
I think there will be env creation failures without the validation for customers who doesn't have extremely specific network setup and in this case they need to ask for an entitlement to have the validation on their env creations. I believe the better way is the opposite and have the validation by default but with entitlement we could disable it for customers with specific network setups.

Reply:
I think disable the validation by default is not a good idea because we preffer to prevent problems, with this solution we potentialy hide a problem in case of cluster deployment.

Reply:
@biharitomi @doktoric I sent an email about why this design was chosen last Thursday. I realize that was right before your holidays so you may not have seen it. But we discussed this quite extensively in the semi-private network group, and it was decided that this is the option that makes the most sense right now.

1. The validation is only useful if it can accurately validate the network setup. This logic can't do that, because there isn't currently a way in CB to evaluate if a subnet has a route to the internet outside of its direct route table. That's the logic I tried to add in CB-11070, but it caused all those problems with the firewall service.
2. This is already an advanced network feature, and should only be used by advanced users who understand what it means for a subnet to have a route to the internet.
3. This validation is doing more harm than good. We only have evidence of one customer actively using this feature, and this validation has prevented them from being able to create an environment for over a month.
4. This validation can cause an environment to fail to be created. But if it's not used, then the environment will be created successfully. The worst case scenario in that situation is that the customer didn't use public subnets and so can't access their UIs and APIs from outside their internal network, but they can use every other feature of feature of CDP. So they are significantly less blocked even if they don't set up the public subnets correctly. And I'm going to work with our docs writer to document troubleshooting procedures for that potential scenario.

The reasons you're stating are the reasons I added the validation in the first place. And it's also why it's behind an entitlement, so that we can quickly re-enable it if we need to. But this is the approved design by the people who are responsible for and have worked on this feature.

Reply:
I would preffer 2 possible choice here:
- remove the validation completely 
- create a disable_validation entitlement and only grant for exxon

Reply:
I agree with @doktoric . It doesn't make sense to add a code that is not used. I'm fine with dropping the validation if we can't make it work well/effectively now and we should be able to add it later.

> we can quickly re-enable it if we need to

what is the use-case for this?

Reply:
I have already made the change to make the entitlement disable the validation, as per Richard's second suggestion.